cfgov/login/auth.py

```diff
@@ -0,0 +1,158 @@
+import logging
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group, Permission
+from django.contrib.contenttypes.models import ContentType
+from django.utils.encoding import smart_str
+from django.utils.module_loading import import_string
+
+from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+
+
+User = get_user_model()
+
+logger = logging.getLogger(__name__)
+
+
+def username_from_email(email, claims=None):
+    """Generate a username for new users from their email address
+
+    If this function is called, it's already tried to look up an email
+    address to match a user and failed. But if the user portion of the email
+    address already exists as a user, this function will try to find an
+    unused username by appending a counter to the end."""
+    email_username = smart_str(email.split("@")[0])
+
+    username = email_username
+    counter = 1
+    while User.objects.filter(username=username).exists():
+        username = email_username + str(counter)
+        counter += 1
+
+    return username
+
+
+def process_roles_admin(user, roles):
+    """Adjust a user's is_superuser property based on OIDC roles claim"""
+    user_modified = False
+
```

Comment:
The next few suggestions are just code simplification, the `user_modified` variable doesn't seem to be needed.

```suggestion
```